Xamarin.Forms.Core/MediaElement.cs

```diff
@@ -0,0 +1,267 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_MediaElementRenderer))]
+	public sealed class MediaElement : View
+	{
+		public static readonly BindableProperty AreTransportControlsEnabledProperty =
+		  BindableProperty.Create(nameof(AreTransportControlsEnabled), typeof(bool), typeof(MediaElement), false);
+
+		public static readonly BindableProperty AspectProperty =
+		  BindableProperty.Create(nameof(Aspect), typeof(Aspect), typeof(MediaElement), Aspect.AspectFit);
+
+		public static readonly BindableProperty AutoPlayProperty =
+		  BindableProperty.Create(nameof(AutoPlay), typeof(bool), typeof(MediaElement), true);
+
+		public static readonly BindableProperty BufferingProgressProperty =
+		  BindableProperty.Create(nameof(BufferingProgress), typeof(double), typeof(MediaElement), 0.0);
+
+		public static readonly BindableProperty IsLoopingProperty =
+		  BindableProperty.Create(nameof(IsLooping), typeof(bool), typeof(MediaElement), false);
+
+		public static readonly BindableProperty KeepScreenOnProperty =
+		  BindableProperty.Create(nameof(KeepScreenOn), typeof(bool), typeof(MediaElement), false);
+
+		public static readonly BindableProperty SourceProperty =
+		  BindableProperty.Create(nameof(Source), typeof(Uri), typeof(MediaElement));
+
+		public static readonly BindableProperty CurrentStateProperty =
+		  BindableProperty.Create(nameof(CurrentState), typeof(MediaElementState), typeof(MediaElement), MediaElementState.Closed);
+
+		public static readonly BindableProperty PositionProperty =
+		  BindableProperty.Create(nameof(Position), typeof(TimeSpan), typeof(MediaElement), TimeSpan.Zero, validateValue: ValidatePosition);
+
+		static bool ValidatePosition(BindableObject bindable, object value)
+		{
+			MediaElement element = bindable as MediaElement;
+			if (element != null)
+			{
+				if (element._renderer != null)
+				{
+					element._renderer.Seek((TimeSpan)value);
+				}
+			}
```

Comment:
I think this can be written in one line `element?._renderer?.Seek((TimeSpan)value);`

Reply:
This doesn't follow our normal approach , `Core` should not call a renderer. We need a new approach for this, use a event like `SeekChanged` or something like that that the renderer will subscribe. 

But why don0t we just check PropertyChanged on the renderer for this what validation is done here that can't be done on the renderer ?

Reply:
I agree, `Core` should not know about the renderer.